### PR TITLE
HPCC-15841 Clarify Logical Filename Rules

### PR DESCRIPTION
**Pull request by JamesDeFabia**
Signed-off-by: Jim DeFabia <jamesdefabia@lexisnexis.com>

@jakesmith Please review 

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

File: docs/ECLLanguageReference/ECLR_mods/Recrd-ScopeNLogicalFilenames.xml [hunk 1]
```diff
@@ -13,16 +13,26 @@
...
-    override the default scope name for the cluster.</emphasis></para>
-
-    <para>For example, assuming you are operating on a cluster whose default
-    scope name is “Training” then the following two OUTPUT actions result in
-    the same scope:</para>
+    <para>The logical filenames used in
+    DATASET and INDEX attribute definitions and the OUTPUT and BUILD (or
+    BUILDINDEX) actions can optionally begin with a ~ meaning it is absolute,
+    otherwise it is relative (the platform configured scope prefix is
+    prepended). It may contain directories. The scope and directories are
```
Comment (jakesmith) [inline docs/ECLLanguageReference/ECLR_mods/Recrd-ScopeNLogicalFilenames.xml]: Not sure appropriate to mention directories here.
Scopes happen to map to directories in our current implementation, but in theory could not. I think it's better just to talk about scopes here.

The exception is in a physical file specification (i.e. ~file::...), but you have a dedicated section for those under the heading of "Landing_Zone_Files". 
Similarly you don't mention ~foreign syntax in this section, as you have a dedicated section for it.

**Comment by jakesmith** [comment]
@JamesDeFabia - looks cleared, but one comment.

File: docs/ECLLanguageReference/ECLR_mods/Recrd-ScopeNLogicalFilenames.xml (hunk 1 above)
Comment (JamesDeFabia) [inline docs/ECLLanguageReference/ECLR_mods/Recrd-ScopeNLogicalFilenames.xml]: @jakesmith Edits made, please have another look

**Comment by jakesmith** [comment]
@JamesDeFabia - looks good.

**Pull request merged**